### ISSUE TITLE
How to query multiple pod names?

### ISSUE THREAD

**Issue by hengfeiyang** [openobserve/openobserve#887]
### Which OpenObserve functionalities are relevant/related to the feature request?

log search

### Description

Like this screenshot:
![image](https://github.com/openobserve/openobserve/assets/1628250/4522cfd8-b307-4edd-8703-113b6c9de759)

I want to query multiple `pod_name`, but in the UI, I must click multiple times `+` and then change the query from `and` to `or`, Can we support multiple select? I remember `Datadog` can do that. they have a checkbox for each field.

### Proposed solution

add a checkbox to let user select multiple values

### Alternatives considered

no

**Comment by prabhatsharma** [comment]
It will be difficult to implement OR, NOT in a checkmark interface.

**Comment by gaby** [comment]
In Kibana you can use wildcards to search within fields. That doesnt work in OpenObserve. For example in @hengfeiyang query you could do `kuberneted_pod_name: '*-center-api-*'` but its not supported by OpenObserve.

Kibana: https://www.elastic.co/guide/en/kibana/current/kuery-query.html#_filter_for_documents_using_wildcards

This is also a bit related to: https://github.com/openobserve/openobserve/issues/708

**Comment by hengfeiyang** [comment]
Yes, we are not ES, you can do it like this in OpenObserve:
```sql
select * from table where str_match(kuberneted_pod_name, '-center-api-')
```

**Comment by prabhatsharma** [comment]
> In Kibana you can use wildcards to search within fields. That doesnt work in OpenObserve. For example in @hengfeiyang query you could do `kuberneted_pod_name: '*-center-api-*'` but its not supported by OpenObserve.
> 
> Kibana: https://www.elastic.co/guide/en/kibana/current/kuery-query.html#_filter_for_documents_using_wildcards
> 
> This is also a bit related to: #708

When it comes to OpenObserve, think in SQL.

another way in addition to what Hengfei mentioned is:

```sql
select * from table where kuberneted_pod_name like '%-center-api-%'
```

**Comment by gaby** [comment]
Thanks, just wish there was ways to do this natively within the UI.

**Comment by prabhatsharma** [comment]
This gives me an idea. We already have `=` and `!=` signs on the field list. We could add `~` sign. This would add `like` or `str_match` for string types. Any ideas on what possible behavior `~` might have on int/float and date.

**Comment by prabhatsharma** [comment]
> add a checkbox to let user select multiple values

@hengfeiyang Now that I think this actually makes sense. We should give this a shot.

**Comment by gaby** [comment]
@prabhatsharma @hengfeiyang I ran into a use case where using `str_match` doesnt work. 

Let's say I have a lot of containers nodes sending logs to O2. But I want to filter only for gateways `x.x.0.1`. For example:

- 10.0.0.1
- 10.1.0.1
- 10.2.0.1
- 10.n.0.1

There's no way to query this using O2 SQL or str_match. In Kibana you can do `hostname == "10.*.0.1"`

I can see this limitation also applying to container names.

**Comment by hengfeiyang** [comment]
You can try:
```sql
re_match(hostname, '10.*.0.1')
```
instead of:
```sql
str_match(hostname, '0.1')
```

https://openobserve.ai/docs/functions/

**Comment by gaby** [comment]
I didnt know `re_match` was a thing! Thanks will give a try.

**Comment by prabhatsharma** [comment]
you could always use standard sql `like` too

**Comment by gaby** [comment]
@prabhatsharma The problem is this forces me to write SQL, which I hate and is one of the biggest issues with O2. 

Looking at the re_match, while it solves the problem. It is not an option when dropping a Filter Field in the Dashboards. There's also no way to use the same field multiple times as a filter

**Comment by prabhatsharma** [comment]
Yeah, we are thinking of building a UI to use SQL functions and operators in dashoards. Let us know if you have ideas around this.

**Comment by gaby** [comment]
@hengfeiyang Tested with `re_match(hostname, '10.*.0.1')` and it doesn't work. Example data:

```
10.1.0.1
10.2.0.1
10.100.0.1
10.1.0.100
10.2.0.123
```

All of them match when using `re_match(hostname, '10.*.0.1')`. I tried doing this `re_match(hostname, '10.*.0.1$')` but I get a SQL failure. Also tried `re_match(hostname, '10.([0-9]+).0.1')` and it doesn't match.
